### ISSUE TITLE
Integrating DeNS in OCP 1.2

### ISSUE THREAD

**Issue by IliasChair** [atomicarchitects/DeNS#3]
Hello everyone,

I have a quick question regarding the integration of DeNS into the OCP repo. The version of the OCP repo currently referenced in the DeNS docs is outdated, and I’ve encountered stability and performance issues while trying to get it running.

Are there any plans to integrate the DeNS architecture into the EquiformerV2 model in the current OCP repo? Given that DeNS is under the MIT license, would it be possible for me to update and integrate the DeNS code myself (time permitting)? I’d be happy to submit a pull request if needed.

On a technical note, I’ve noticed the code distinguishes between OC20 'all' and 'MD' splits. Why is this distinction necessary? Shouldn't Equiformer/DeNS function independently of the dataset (aside from the training config of course)?

Thanks for your time, and I look forward to hearing your thoughts.

Best,
Ilias

**Comment by yilunliao** [comment]
Hi @IliasChair 

Thanks for your interest! Please see my responses below.

---

> 1. The version of the OCP repo currently referenced in the DeNS docs is outdated, and I’ve encountered stability and performance issues while trying to get it running.

This repo was based on the version before OCP was updated to FAIR-Chem. We remain consistent throughout the project and release the original codebase reproducing the numbers claimed in the paper.

However, I might not be very familiar with their latest codebase since there have been many updates.

---

> 2. Are there any plans to integrate the DeNS architecture into the EquiformerV2 model in the current OCP repo? Given that DeNS is under the MIT license, would it be possible for me to update and integrate the DeNS code myself (time permitting)?

On my side, I would not have enough bandwidth to update this repo to be compatible with the latest OCP (or FAIR-Chem) codebase recently.

But you are encouraged to upgrade and integrate the DeNS codebase (either pull request to this repo or to OCP repo). Feel free to ask more questions if you have.

---

> 3. Why to distinguish between OC20 'all' and 'MD' splits. Why is this distinction necessary? Shouldn't Equiformer/DeNS function independently of the dataset (aside from the training config of course)?

We thought that structures in the MD split have more variations and thus more information. Therefore, overfitting them would require training for longer. Since we trained on OC20 All+MD dataset for only 2 epochs, learning the force labels in the MD split without denoising would be sufficient. In our early stage of experiments, we found that adding noise to structures in the MD split can make models converge slower than training without DeNS and thus restricted DeNS to only structures in the All split.

DeNS would slightly depend on datasets. If the datasets are too hard or models cannot overfit the datasets (although there are other reasons like weaker models or training for too short amounts of time), using DeNS as additional regularization would not result in much performance gain.

**Comment by IliasChair** [comment]
Hi @yilunliao 
Thank you so much for the swift and detailed response!
I’ll start working on updating and integrating DeNS into the latest OCP repo myself. Given the amount of effort involved, it’s hard to say right now if I’ll be able to fully complete it, but I’ll give it a shot.

Thanks again for your time and support. I might reach out with more questions as I work through the integration. I’ll go ahead and close this issue for now and reopen it if any new questions come up.

Best regards,
Ilias
